packages/lodestar/src/chain/blocks/process.ts

```diff
@@ -126,27 +131,47 @@ export function updateForkChoice(
 }
 
 export async function runStateTransition(
-  config: IBeaconConfig,
-  db: IBeaconDb,
-  logger: ILogger,
-  stateContext: Required<IStateContext>,
+  emitter: ChainEventEmitter,
+  stateContext: ITreeStateContext,
   job: IBlockProcessJob
 ): Promise<IStateContext | null> {
+  const config = stateContext.epochCtx.config;
+  const {SLOTS_PER_EPOCH} = config.params;
+  const {BeaconBlockHeader} = config.types;
   try {
+    const preSlot = stateContext.state.slot;
+    const postSlot = job.signedBlock.message.slot;
+    const preEpoch = computeEpochAtSlot(config, preSlot);
+    let nextEpochSlot = computeStartSlotAtEpoch(config, preEpoch + 1);
+    while (nextEpochSlot < postSlot) {
+      processSlots(stateContext.epochCtx, stateContext.state, nextEpochSlot);
+      const blockHeader = BeaconBlockHeader.clone(stateContext.state.latestBlockHeader);
+      blockHeader.stateRoot = stateContext.state.hashTreeRoot();
```

Comment:
it seems that we should only set `stateRoot` if it's ZERO_HASH https://github.com/ChainSafe/lodestar/blob/master/packages/lodestar-beacon-state-transition/src/slot.ts#L32

I think checkpoint `root` is the same for this whole while loop because these are skipped slots, only checkpoint epoch is different. It means checkpoint `root` can even be calculated before this `while` loop.